visualizer/semanticsEditor.js

```diff
@@ -45,25 +48,34 @@
 
   // Creates semantics editor result container and fills it the with `resultBlock`,
   // each of which reprensents a semantic result on this node.
-  function createAndLoadSemanticResult(traceNode, selfWrapper) {
+  function createAndLoadResultContainer(traceNode, selfWrapper) {
     var resultContainer = domUtil.createElement('.result');
     var results = ohmEditor.semantics.getResults(traceNode);
     if (!results) {
       return resultContainer;
     }
+    var idx = 0;
     Object.keys(results).forEach(function(opName) {
       var resultList = results[opName];
       resultList.forEach(function(resultWrapper) {
-        resultContainer.appendChild(createResultBlock(opName, resultWrapper));
-        if (resultWrapper.isNextStep) {
+        var resultBlock = resultContainer.appendChild(createResultBlock(opName, resultWrapper));
+        // If the result block is not the first one that contains a result, then add a left border
+        // to it to seperate it from its former block
+        if (resultBlock.textContent) {
+          resultBlock.classList.toggle('leftBorder', idx++ > 0);
+        }
+        if (!resultWrapper.forced && resultWrapper.isNextStep) {
           selfWrapper.classList.add('nextStep');
         }
-        selfWrapper.classList.toggle('passThrough', resultWrapper.isPassThrough);
+
         if (resultWrapper.forCallingSemantic) {
           selfWrapper._args = resultWrapper.args;
+          selfWrapper.classList.toggle('passThrough', !!resultWrapper.isPassThrough);
```

Comment:
Nit: do you really need the coercion (`!!`) here? `isPassthrough` already implies that it's a boolean -- so if necessary, you should probably do the coercion when you set `isPassthrough`.